### PR TITLE
[PATCH v5] api: packet: change odp_packet_reset() specification

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

**Comment by psavol** [comment]
commit log typos: specifications -> specification

File: include/odp/api/spec/packet.h [hunk 1]
```diff
@@ -123,20 +123,38 @@ void odp_packet_free_sp(const odp_packet_t pkt[], int num);
...
- * to initialize pointers and lengths. It must be less than the total buffer
- * length of the packet. Packet is not modified on failure.
+ * Resets all packet metadata and adjusts packet data to start according to pool
+ * parameters (align, seg_len, headroom). Packet data length is set to 'len'.
+ * Maximum value for 'len' can be queried with odp_packet_reset_max_len()
+ * function.
+ *
+ * Function maintains the original user area contents. Packet data persistence
+ * is not guaranteed and possible extra packet segments may be freed by the
+ * implementation. Packet is not modified on failure.
```
Comment (psavol) [inline include/odp/api/spec/packet.h:133]: Suggesting slight change of wording on this chapter:

"Packet reset maintains the original user area content. The operation may change data layout in packet segments and free possible extra segments. Data content may not be preserved. ..."

File: include/odp/api/spec/packet.h [hunk 2]
```diff
@@ -155,6 +155,25 @@ int odp_packet_reset(odp_packet_t pkt, uint32_t len);
  * @return Maximum packet data length for odp_packet_reset()
  */
 uint32_t odp_packet_reset_max_len(odp_packet_t pkt);
+
+/**
+ * Reset packet metadata
+ *
+ * Resets all packet metadata, except the ones listed below, to their default
+ * values:
```
Comment (psavol) [inline include/odp/api/spec/packet.h:163]: Suggesting a simplified sentence:

"Resets all packet metadata to their default values, except the ones listed below:"

File: include/odp/api/spec/packet.h (hunk 1 above)
Comment (MatiasElo) [reply include/odp/api/spec/packet.h]: Changed in V2.

File: include/odp/api/spec/packet.h (hunk 2 above)
Comment (MatiasElo) [reply include/odp/api/spec/packet.h]: Changed in V2.

**Comment by MatiasElo** [comment]
V2:
- Petri's comments
- Improved tests

**Review by psavol — APPROVED** [review]

**Comment by ashwinyes** [comment]
Acked-by: Ashwin Sekhar T K <asekhar@marvell.com>

File: include/odp/api/spec/packet.h [hunk 3]
```diff
@@ -123,20 +123,38 @@ void odp_packet_free_sp(const odp_packet_t pkt[], int num);
 /**
  * Reset packet
  *
- * Resets all packet metadata to their default values. Packet length is used
- * to initialize pointers and lengths. It must be less than the total buffer
- * length of the packet. Packet is not modified on failure.
+ * Resets all packet metadata and adjusts packet data to start according to pool
+ * parameters (align, seg_len, headroom). Packet data length is set to 'len'.
+ * Maximum value for 'len' can be queried with odp_packet_reset_max_len()
+ * function.
```
Comment (psavol) [inline include/odp/api/spec/packet.h:129]: To be clear, it's allowed to pass len >max_len (will return failure). Len <= max_len will succeed.

**Pull request merged**